includes/wikia/services/DataMartService.class.php

```diff
@@ -707,100 +638,38 @@ public static function getTopCrossWikiArticlesByPageview( $hub, $langs, $namespa
-	 * @param string $langCode A valid Wiki's language code
-	 * @param integer $periodId
-	 * @param integer $limit [OPTIONAL] The maximum number of items in the list, defaults to 200
-	 *
-	 * @return Array The list, the key contains Wiki ID's and "pageviews" number
-	 */
-	public static function getTopTagsWikisByPageviews ( $tagId, $startDate, $endDate, $langCode = 'en', $periodId = null, $limit = 200 ) {
-		$app = F::app();
-
-		if ( empty( $endDate ) ) {
-			if ( $periodId == self::PERIOD_ID_MONTHLY ) {
-				$endDate = date( 'Y-m-01' );
-			} else {
-				$endDate = date( 'Y-m-d', strtotime( '-1 day' ) );
-			}
-
-		}
-
-		if ( empty( $periodId ) ) {
-			$periodId = self::PERIOD_ID_MONTHLY;
-		}
-
-		$memKey = wfSharedMemcKey( 'datamart', 'tags_top_wikis', $tagId, $periodId, $startDate, $endDate, $langCode, $limit );
-		$tagViews = $app->wg->Memc->get( $memKey );
-		if ( !is_array( $tagViews ) ) {
-			$tagViews = array();
-			if ( !self::isDisabled() ) {
-				$db = DataMartService::getDB();
-
-				$tables = array(
-					'r' => 'rollup_wiki_pageviews',
-					'c' => 'wikicities.city_tag_map',
-					'd' => 'dimension_wikis'
-				);
-				$fields = array(
-					'c.tag_id as tag_id',
-					'r.wiki_id',
-					'sum(r.pageviews) as pviews'
-				);
-				$cond = array(
-					'period_id' => $periodId,
-					'tag_id' => $tagId,
-					"time_id between '{$startDate}' AND '{$endDate}'",
-				);
-				$opts = array(
-					'GROUP BY' => 'c.tag_id, r.wiki_id',
-					'ORDER BY' => 'pviews DESC',
-					'LIMIT' => $limit
-				);
-				$join_conds = array(
-					'c' => array( 'INNER JOIN', array( 'c.city_id = r.wiki_id' ) ),
-					'd' => array( 'INNER JOIN', array( 'd.wiki_id = r.wiki_id', 'd.public = 1', "d.lang = '{$langCode}'" ) )
-				);
-
-				$result = $db->select( $tables, $fields, $cond, __METHOD__, $opts, $join_conds );
-
-				while ( $row = $db->fetchObject( $result ) ) {
-					$tagViews[$row->wiki_id] = $row->pviews;
-				}
-
-				$app->wg->Memc->set( $memKey, $tagViews, 60 * 60 * 12 );
-			}
-		}
-
-		return $tagViews;
-	}
-
-	/**
 	 * Gets page views for given articles
 	 *
 	 * @param array $articlesIds
 	 * @param datetime $timeId
+	 * @param int $wikiId
+	 * @param int $periodId
 	 * @return array
 	 */
 	public static function getPageViewsForArticles( Array $articlesIds, $timeId, $wikiId, $periodId = self::PERIOD_ID_WEEKLY ) {
-		$db = DataMartService::getDB();
+		try {
+			$db = DataMartService::getDB();
 
-		$articlePageViews = ( new WikiaSQL() )->skipIf( self::isDisabled() )
-			->SELECT( 'article_id', 'pageviews' )
-			->FROM( 'rollup_wiki_article_pageviews' )
-			->WHERE( 'article_id' )->IN( $articlesIds )
-			->AND_( 'time_id' )->EQUAL_TO( $timeId )
-			->AND_( 'wiki_id' )->EQUAL_TO( intval( $wikiId ) )
-			->AND_( 'period_id' )->EQUAL_TO( intval( $periodId ) )
-			->runLoop( $db, function( &$articlePageViews, $row ) {
-				$articlePageViews[ $row->article_id ] = $row->pageviews;
-			} );
+			$articlePageViews =
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$articlePageViews". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)